lib/ansible/executor/stats.py

```diff
@@ -91,6 +94,7 @@ def update_custom_stats(self, which, what, host=None):
 
         # mismatching types
         if not isinstance(what, type(self.custom[host][which])):
+            display.warning("Mismatching types in stats")
```

Comment:
i would add which,what and host information, narrowing down where the mismatch hapened